### PR TITLE
Write log messages smaller than 2048 character systemd limit

### PR DESCRIPTION
**Pull request by Merlin-Taylor**
This change reduces the size of log messages imposed by a [systemd limitation](https://bugs.freedesktop.org/show_bug.cgi?id=86465) that splits log lines at 2048 characters. This renders long JSON messages un-parseable and prevents them from being propagated along our [logging pipeline](https://wiki.thetrainline.com/display/ELK/Linux+logging)

This change reduces the length of stack traces and removes less informative fields from log messages.

[Example HTTP error message](https://kibana.thetrainline.com/#/doc/logstash-prod-linux-*/logstash-prod-linux-2017.02.22/syslog?id=AVpl8YHvU8zsl8V2hrD-&_g=(refreshInterval:(display:Off,pause:!f,section:0,value:0),time:(from:now-15m,mode:quick,to:now))) Note that the stack trace, URL, swagger parameters and user are all displayed.

https://jira.thetrainline.com/browse/PD-133

### CODE REVIEW

File: server/modules/logger.js [hunk 1]
```diff
@@ -8,27 +8,20 @@ let AWS = require('aws-sdk');
 
 let config = require('config/');
 const EM_LOG_LEVEL = config.get('EM_LOG_LEVEL').toLowerCase();
+let fp = require('lodash/fp');
 
 // Set up formatter - if local, leave undefined
 let formatter;
-if (config.get('IS_PRODUCTION') === true) {
+if (true) {
```
Comment (jake-cross) [inline server/modules/logger.js]: if(true && "true" && 1) {
  console.log('What is this?');
}

File: server/modules/miniStack.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        };
+      } else if (inOtherCode(line, prev)) {
+        return {
+          context: contextLines ? [line] : [],
+          omittedCount: omittedCount + 1,
+          output,
+          prev: line
+        };
+      } else {
+        throw new Error('This code path should be unreachable');
```
Comment (jake-cross) [inline server/modules/miniStack.js]: I've seen other PR's with Error messages extracted to their own class. Not too concerned this second about it not being there - but maybe we should pick a standard as a team.

File: server/modules/logger.js (hunk 1 above)
Comment (DavidIHunt) [inline server/modules/logger.js]: Yeah.. That's... What???

File: server/modules/miniStack.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    }
+    let { context, omittedCount, output } = lines.reduce(loop, {
+      context: [],
+      omittedCount: 0,
+      output: [],
+      prev: undefined
+    });
+    let skipped = linesSkippedMessage(omittedCount);
+    let rest = context.length > 0 ? skipped.concat(context.map(shorten)) : [];
+    return output.concat(rest).join('\n');
```
Comment (jake-cross) [inline server/modules/miniStack.js]: I still have to force myself to think "is there an ES6 way to do this?" but on that note, I like (...myArray) over myArray.concat()

**Comment by DavidIHunt** [comment]
LGTM

**Pull request merged**